### ISSUE TITLE
hparam_search: replace toy_model.py with an opaque callable

### ISSUE THREAD

**Issue by hyang0129** [hyang0129/onlycodes#168]
## Problem

`iterative_numerical/hparam_search/workspace/toy_model.py` ships the full scoring formula in source the agent can read:

```python
lr_score = math.exp(-3.0 * (math.log10(learning_rate / 0.01)) ** 2)
hs_score = math.exp(-((hidden_size - 128) / 128.0) ** 2)
do_score = math.exp(-((dropout - 0.3) / 0.2) ** 2)
return round(0.95 * lr_score * hs_score * do_score, 6)
```

Three independent Gaussian peaks at `lr=0.01, hs=128, dropout=0.3`. An agent that reads this file (which the prompt requires — `import evaluate from toy_model`) infers the optimum by inspection in seconds. No search loop is needed. The grader's `_evaluate` is byte-identical to `toy_model.py`, so the answer just IS in the workspace.

This defeats the stated goal — the prompt frames the task as "start with a coarse grid search, then refine."

## Acceptance criteria

Pick one of these approaches:

- [ ] **Opaque callable** — ship a compiled / bytecoded / precomputed function the agent can only invoke, not read. E.g. a frozen pickle, a lookup table backed by a JSONL with hashed keys, or a `.pyc` only.
- [ ] **Per-instance peaks** — derive the Gaussian peaks from `sha256(instance_id)` so the formula is the same shape but the optimum varies per task instance. Document this in `prompt.md` so the agent knows to search rather than read.
- [ ] **Add cross terms / lookup table** — make the function not analytically invertible (e.g. `score = base_table[round(lr*1000), round(hs/8), round(do*10)]` where `base_table` is committed as data, not code).

Whichever option is chosen:
- [ ] Update `prompt.md` so it does not promise the agent can solve by inspection.
- [ ] Update `grader/hidden.py` to reference the same opaque mechanism (so reference and grader still agree).
- [ ] Confirm `reference_output.json` still passes `grade(reference) → passed=True, score=1.0`.
- [ ] Add at least one test where the global optimum is *not* near `(0.01, 128, 0.3)` to prove search is being exercised.
